### PR TITLE
Bugfixes in Netbox inventory

### PR DESCRIPTION
**Pull request by aruhier**
##### SUMMARY

Fix authentication and python 3 incompatibilities in Netbox dynamic inventory. Also did some cleanup.

Related to #45347

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME

Netbox dynamic inventory

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (netbox_virtual a47eefad88) last updated 2018/09/17 15:01:38 (GMT +200)
  config file = None
  configured module search path = [u'/Users/sieben/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/sieben/workspace/ansible/lib/ansible
  executable location = /Users/sieben/workspace/ansible/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### ADDITIONAL INFORMATION

@sieben: could you please also review this PR, please?

### CODE REVIEW

**Comment by remyleone** [comment]
LGTM

**Comment by remyleone** [comment]
@Anthony25 I think it is related to your commit message: Working with @sieben on this

File: .github/BOTMETA.yml [hunk 1]
```diff
@@ -855,7 +855,7 @@ files:
       - inventory
     labels:
       - cloud
-  lib/ansible/plugins/inventory/netbox.py: sieben
+  lib/ansible/plugins/inventory/netbox.py: sieben Anthony25
```
Comment (pilou-) [inline .github/BOTMETA.yml]: There is no need to add module authors in botmeta (by default, module authors are maintainers), remove the whole entry.

**Comment by remyleone** [comment]
@gundalow Could you merge this?

**Comment by remyleone** [comment]
LGTM

**Comment by remyleone** [comment]
@maxamillion Could you merge this?

**Comment by gundalow** [comment]
Does this need backporting to `stable-2.7`?

**Comment by remyleone** [comment]
Yes otherwise it won't work with NetBox instances that requires authentication

**Comment by gundalow** [comment]
Actually, I don't see this code in `stable-2.7` https://github.com/ansible/ansible/tree/stable-2.7/lib/ansible/plugins/inventory

So I don't think their is anything to backport

**Comment by aruhier** [comment]
Thanks!

**Pull request merged**